ext/reflection/php_reflection.c

```diff
@@ -1334,6 +1335,10 @@ static reflection_type_kind get_type_kind(zend_type type) {
 	}
 
 	if (ZEND_TYPE_IS_COMPLEX(type)) {
+		/* BC support for 'iterable' type */
+		if (UNEXPECTED(ZEND_TYPE_IS_ITERABLE_FALLBACK(type))) {
```

Comment:
This could be correct or not, depending which type `_ZEND_TYPE_ITERABLE_BIT` is set on the top type which could also be a union.

Reply:
Yeah, this comment was from the time where full BC compatibility in terms of reflection was aimed for.